MigrationGuide.md

```diff
@@ -0,0 +1,224 @@
+## coreMQTT version >=v2.0.0 Migration Guide
+
+With coreMQTT versions >=v2.0.0, there are some breaking changes that need to be addressed when upgrading.
+
+### Breaking Changes
+
+The `MQTT_ProcessLoop` function no longer uses a timeout as this led to unavoidable busy-waiting. Thus, the signature of `MQTT_ProcessLoop` changed from `MQTTStatus_t MQTT_ProcessLoop( MQTTContext_t * pContext, uint32_t timeoutMs )`  to  `MQTTStatus_t MQTT_ProcessLoop( MQTTContext_t * pContext )`. Additionally, `MQTT_ProcessLoop` can now return `MQTTNeedMoreBytes`. A return of `MQTTNeedMoreBytes` means that `MQTT_ProcessLoop` received only a part of an MQTT packet and will need to be called again (probably after a bit of delay) in order to finish receiving the MQTT packet. Thus, to migrate, simply remove the timeout from the `MQTT_ProcessLoop` call and additionally check for if `MQTTNeedMoreBytes` has been returned when checking the status of `MQTT_ProcessLoop`. For example:
```

Comment:
Since these changes are not numbered (or added to bullets). They look like a big monolithic blob of text. I think we should try to separate them somehow.

Ref: take a look at the rendered markdown file here: https://github.com/FreeRTOS/coreMQTT/blob/coreMQTTMigration/MigrationGuide.md

Reply:
added bullets